### ISSUE TITLE
[Docs] Live example for the Lifecycle Hooks not working as expected.

### ISSUE THREAD

**Issue by jainanuj0812** [angular/angular#22279]
[X ] Documentation issue or request

## Current behavior
https://angular.io/guide/lifecycle-hooks
Live example for lifecycle hooks in documentation is not working as expected.

Below is the example link
https://stackblitz.com/angular/jlqeyrgaoaq

Clicking the 'Create PeekABooComponent' not logging the lifecycle hooks below it on UI.

## Expected behavior
After clicking the 'Create PeekABooComponent' the logs should be viewed below it.

Possible Solution

The **toggleChild** method of PeekABooParentComponent should be:-

```javascript
  toggleChild() {
    this.hasChild = !this.hasChild;
    if (this.hasChild) {
      this.heroName = 'Windstorm';
      this.logger.clear(); // clear log on create
    }
    **this.hookLog = this.logger.logs();**
    this.logger.tick();
  }
```

**Comment by jainanuj0812** [comment]
Hi @brandonroberts, I have it on my local, will push it by EOD or by tomorrow.

**Comment by brandonroberts** [comment]
Great, thanks

**Comment by gkalpak** [comment]
Closing since the original issue should have been fixed by #23201.
